### PR TITLE
Record click events for unlock and auth code checks.

### PR DESCRIPTION
**Pull request by delputnam**
We want to record when a user clicks on the buttons to check the unlock state and the auth code checks when on the pre-check page to see where users may be having trouble during an inbound transfer.

Testing:
- Start a transfer with a domain in a locked state.
- Click on the "I've unlocked my domain" button a few times.
- Then unlock your domain and enter an auth code and click on the "Check my authorization code" button a few times.

- Wait a few minutes and search for the `calypso_transfer_domain_precheck_%_check_click` actions in tracks.

### CODE REVIEW

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 1]
```diff
@@ -90,11 +91,17 @@ class TransferDomainPrecheck extends React.Component {
 	};
 
 	refreshStatus = () => {
-		this.props.refreshStatus( this.statusRefreshed );
+		this.props.refreshStatus( this.statusRefreshed ).then( result => {
+			const isUnlocked = get( result, 'inboundTransferStatus.unlocked' );
+			this.props.recordUnlockedCheckButtonClick( this.props.domain, isUnlocked );
```
Comment (delputnam) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: I'm not sure why calling `recordUnlockedCheckButtonClick` inside of `then()` causes this error:

```Uncaught (in promise) TypeError: Cannot read property 'type' of undefined
    at eval (wpcom-api-middleware.js:82)
    at eval (index.js:12)
    at Object.eval [as recordUnlockedCheckButtonClick] (redux.js:457)
    at eval (transfer-domain-precheck.jsx:125)

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 2]
```diff
@@ -90,11 +91,17 @@ class TransferDomainPrecheck extends React.Component {
 	};
 
 	refreshStatus = () => {
-		this.props.refreshStatus( this.statusRefreshed );
+		this.props.refreshStatus( this.statusRefreshed ).then( result => {
+			const isUnlocked = get( result, 'inboundTransferStatus.unlocked' );
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: What if this is an error? It's going to be `undefined`. Might want to treat that that differently.
screen shot 2018-08-04 at 22 42 25

Comment (delputnam) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Good catch. Thanks!

**Comment by delputnam** [comment]
I've added `checkLockedStatus` to record the clicks when `this.props.unlocked` any value besides `false` before moving to the next step.

I've also translated an unlocked status of `null` to `"unavailable"` in the logging function which is what a return of `null` actually means...it means that the endpoint couldn't get any whois status information for that domain.

If the transfer status endpoint returns an error, the the user will never get to the precheck screen, but assuming that somehow the user gets to that page and then the endpoint starts returning an error, tracks will record `is_unlocked` as `undefined` which would mean that an error occurred in the endpoint as opposed to `unavailable` which would mean that there was some other issue with retrieving the whois information.

**Comment by delputnam** [comment]
FYI, the auth code check will also log `undefined` when an error is returned.

**Pull request merged**